File: rust/src/models/metric_filter.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::metric_filter_operator::MetricFilterOperator;

/// A pre-aggregation filter: only events whose `property` matches feed the metric's
/// aggregation. Distinct from a segmentation dimension (which splits pricing). Multiple
/// filters are ANDed.
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct MetricFilter {
    pub op: MetricFilterOperator,

    pub property: String,

    pub values: Vec<String>,
}

impl MetricFilter {
    pub fn new(op: MetricFilterOperator, property: String, values: Vec<String>) -> Self {
        Self {
            op,
            property,
            values,
        }
    }
}

File: typescript/src/models/metricFilter.ts
// this file is @generated
import {
  type MetricFilterOperator,
  MetricFilterOperatorSerializer,
} from "./metricFilterOperator";
/**
 * A pre-aggregation filter: only events whose `property` matches feed the metric's
 * aggregation. Distinct from a segmentation dimension (which splits pricing). Multiple
 * filters are ANDed.
 */
export interface MetricFilter {
  op: MetricFilterOperator;

  property: string;

  values: string[];
}

export const MetricFilterSerializer = {
  _fromJsonObject(object: any): MetricFilter {
    return {
      op: MetricFilterOperatorSerializer._fromJsonObject(object["op"]),
      property: object["property"],
      values: object["values"],
    };
  },

  _toJsonObject(self: MetricFilter): any {
    return {
      op: MetricFilterOperatorSerializer._toJsonObject(self.op),
      property: self.property,
      values: self.values,
    };
  },
};
